src/copy.c

```diff
@@ -324,6 +324,7 @@ TSCopyMultiInsertBufferFlush(TSCopyMultiInsertInfo *miinfo, TSCopyMultiInsertBuf
 	ChunkInsertState *cis =
 		ts_chunk_dispatch_get_chunk_insert_state(miinfo->ccstate->dispatch,
 												 buffer->point,
+												 buffer->slots[0],
```

Comment:
This looks incorrect, we should have decompressed everything we needed. Let's move decompression out of this function? It looks error-prone in cases like this one, when you just want to re-create the insert state, but don't actually want to decompress anything. Or when you use some batching, get the chunk insert state only once, but then have to call decompression for every row.

Reply:
Hmm in the presence of unique constraints you have to do this check for every tuple. Cause different tuples routed to the same chunk might still require different batches to be decompressed. Where would you move this to?

Reply:
To `chunk_dispatch_exec` and `copyfrom`, just after they look up the insert state. And here, in `TSCopyMultiInsertBufferFlush`, you flush the multi-insert buffers with tuples that you have already seen in `copyfrom`, so you don't have to do decompression for them again.